app/controllers/idv/by_mail/request_letter_controller.rb

```diff
@@ -87,6 +87,8 @@ def first_letter_requested_at
         current_user.gpo_verification_pending_profile&.gpo_verification_pending_at
       end
 
+      # So some of these things check constituent parts and take action.
+      # Does that kinda undermine the point of some of this?
       def confirm_mail_not_rate_limited
         redirect_to idv_enter_password_url if gpo_mail_service.rate_limited?
       end
```

Comment:
I.e., `before_action`s confirm this, and that the profile is not too old. I think we just let it be, but noting just for discussion.

Reply:
I actually think this should use the service. We should probably have a before action that confirms that we can send a letter that looks like this:

```ruby
def letter_request_allowed?
  policy = Idv::GpoVerifyByMailPolicy.new(current_user)
  if resend_requested?
    policy.resend_letter_available?
  else
    policy.send_letter_available?
  end
end
```

Then we can remove some of those before actions that are doing the one-off checks.